tools/windows-node-installer/pkg/cloudprovider/aws/ec2.go

```diff
@@ -273,9 +273,82 @@ func (a *awsProvider) findOrCreateSg(infraID string, vpc *ec2.Vpc) (string, erro
 	if err != nil {
 		return a.createWindowsWorkerSg(infraID, vpc)
 	}
+
+	log.V(0).Info(fmt.Sprintf("Using existing Security Group: %s.", sgID))
+
+	// Check winrm port open status for the exist security group
+	portOpenResult, err := a.checkSgWinrmPort(sgID)
+	if err != nil {
+		return "", err
+	}
+
+	// Add winrm port to security group if it doesn't exist
+	if !portOpenResult {
+		err := a.createWinrmPortToSg(sgID)
+		if err != nil {
+			return "", err
+		}
+		log.V(0).Info(fmt.Sprintf("Winrm port is now added to Security Group %s", sgID))
+	} else {
+		log.V(0).Info(fmt.Sprintf("Winrm port already opened for Security Group: %s.", sgID))
+	}
+
 	return sgID, nil
 }
 
+// This function checks whether the winrm https port is opened in the given security group.
```

Comment:
niit: Look at the existing function's comment, we should always start with functionName like
`checkSgWinrmPort checks whether...`